src/main/java/synapseawsconsolelogin/Auth.java

```diff
@@ -321,17 +317,26 @@ public String getAWSSecretKey() {
 		}
 	}
 	
-	private static Properties properties = null;
-
-	public static void initProperties() {
+	public void initProperties() {
 		if (properties!=null) return;
 		properties = new Properties();
+		
+		String propertyFileName = System.getenv(PROPERTIES_FILENAME_PARAMETER);
```

Comment:
sorry i didn't catch this on the previous review.  The `PROPERTIES_FILENAME_PARAMETER` is not a required config however it seems like this would cause an exception if the parameter is not defined?

Reply:
No, System.getenv() does not throw an exception if the passed parameter is not defined.  It *does* throw an exception if the input is null (`System.getenv(null)`).  More here: https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#getenv(java.lang.String)

Reply:
ahh yes, i read that previously but interpreted it incorrectly.  thanks for the java lesson.